code/game/objects/effects/decals/cleanable/food.dm

```diff
@@ -41,6 +42,17 @@
 	if(is_species(AM, /datum/species/snail))
 		to_chat(AM, "<span class='danger'>Your path is obstructed by <span class='phobia'>salt</span>.</span>")
 
+/obj/effect/decal/cleanable/food/salt/Crossed(atom/movable/AM)
+	..()
+	if(isliving(AM))
+		if(iscarbon(AM))
+			var/mob/living/carbon/C = AM
+			if(C.m_intent == MOVE_INTENT_WALK)
+				return
+		safepasses--
+		if(safepasses <= 0 && !QDELETED(src))
+			qdel(src)
```

Comment:
```DM
if(!isliving(AM))
    return
code

Reply:
not sure if this proc needs return values but might be safer to assume return just gives inherit value

`. = ..()`

Reply:
This was posted about further up, it doesn't